kvbc/src/Replica.cpp

```diff
@@ -448,6 +451,41 @@ bool Replica::getBlock(uint64_t blockId, char *outBlock, uint32_t *outBlockSize)
   return true;
 }
 
+std::future<bool> Replica::getBlockAsync(uint64_t blockId, char *outBlock, uint32_t *outBlockSize) {
+  static uint64_t callCounter = 0;
+  static constexpr size_t snapshotThresh = 1000;
+
+  auto future = blocksIOWorkersPool_.async(
+      [this](uint64_t blockId, char *outBlock, uint32_t *outBlockSize) {
+        auto start = std::chrono::steady_clock::now();
+        *outBlockSize = 0;
+
+        // getBlock will throw exception if block doesn't exist
+        try {
+          getBlock(blockId, outBlock, outBlockSize);
+        } catch (NotFoundException &ex) {
+          *outBlockSize = 0;
+          return false;
+        }
+        ConcordAssertGT(*outBlockSize, 0);
+        auto jobDuration =
+            std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::steady_clock::now() - start).count();
+        LOG_DEBUG(logger, "Job done: " << KVLOG(blockId, *outBlockSize, jobDuration));
+        histograms_.get_block_duration->record(jobDuration);
+        return true;
+      },
+      std::forward<decltype(blockId)>(blockId),
```

Comment:
why forwarding is necessary here?

Reply:
yes. Here is what I get without:
`
In file included from /concord-bft/kvbc/src/Replica.cpp:9:
In file included from /concord-bft/kvbc/include/Replica.h:7:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/functional:54:
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/tuple:1662:14: error: no matching function for call to '__invoke'
      return std::__invoke(std::forward<_Fn>(__f),
             ^~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/tuple:1671:19: note: in instantiation of function template specialization
      'std::__apply_impl<std::packaged_task<bool (unsigned long &, char *&, unsigned int *&)> &, std::tuple<unsigned long, char *, unsigned int *>,
      0, 1, 2>' requested here
      return std::__apply_impl(std::forward<_Fn>(__f),
`